### DISCUSSION TITLE
Sensor refresh

### DISCUSSION THREAD

**Discussion by elstevega** [RobertD502/home-assistant-iocare#51] [Q&A] [answered]
I see in the code that the default sensor refresh interval is set to 30 seconds. However, I'm seeing blocks of time where there are no updates for 20-25 minutes. Trying to force an update via Update Entity doesn't seem to do anything.
Any thoughts on this?

thank you

**Comment by RobertD502** [comment]
Turn on debug logging from the integration UI and then restart home assistant.

Once you notice that a sensor isn't updating for a while, you can disable the debug logging. This will download a log file. Compare what the API has been returning (in the debug log) within the timeframe that you suspect the sensor hasn't been updating.

This way you will be able to tell if the API is sending stale data - if so, there is nothing from the integration side that can be done as we don't have control of what goes on on Coway's servers.

You can also disable WiFi on the purifier and then enable it again to force the purifier to reconnect to their servers and hopefully start sending updated data.

**Reply by elstevega** [reply]
Thanks for the prompt reply - I will look into it tomorrow & report back.
cheers!

**Reply by RobertD502** [reply]
If you don't have anything in your logs (prior to using my debug log instructions) about the integration failing to update data, then it 100% is simply the data that is being reported by the Coway API and your only option is to see if you can force the purifier to reconnect to their servers by using the WiFi steps that I outlined.

Out of curiosity, which sensor is the one you're having issues with?

**Reply by elstevega** [reply]
Having trouble with the particulate_matter_10 sensor.
Interestingly, the log files show an update every 10 seconds from 8:47am to 12:38pm. (I modded the source to pool every 10 sec instead of 30 just for kicks)
I'm using the custom:mini-graph card to chart the data & it is showing update intervals as long as 23 minutes - so now I suspect either HA itself, or more likely the card - will try some different charting options to see if I get different results.

Thanks again!

**Reply by RobertD502** [reply, marked as answer]
Click on the entity itself. If the Home Assistant entity graph looks sort of like this then data is being updated:

![image](https://github.com/user-attachments/assets/a17598bb-ea76-4e5f-8ba1-bc7c242b7cbe)

And if you click on `Show more`, then you'll get more details on the history page:

![image](https://github.com/user-attachments/assets/e202eaea-42ce-40f0-9d52-1f7e7df15a91)

**Reply by elstevega** [reply]
It does look like that - I guess I was thrown off by the "13 minutes" ago in the chart.
